File: ios/Sources/RecorderEnginePlugin/RecorderEnginePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(RecorderEnginePlugin)
public class RecorderEnginePlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "RecorderEnginePlugin"
    public let jsName = "RecorderEngine"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = RecorderEngine()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/kotlin/com/xenix/audio/recorder/engine/RecorderEnginePlugin.kt
package com.xenix.audio.recorder.engine;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "RecorderEngine")
class RecorderEnginePlugin: Plugin() {

    private val implementation = RecorderEngine()

    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value") ?: ""
        val ret = JSObject()
        ret.put("value", implementation.echo(value))
        call.resolve(ret)
    }
}
